opm/simulators/linalg/bda/WellContributions.cpp

```diff
@@ -82,22 +125,74 @@ void WellContributions::apply(cl::Buffer& d_x, cl::Buffer& d_y) {
         // copy vector y from CPU to GPU
         queue->enqueueWriteBuffer(d_y, CL_TRUE, 0, sizeof(double) * N, h_y);
     }
+}
+*/
 
+void WellContributions::getParams(unsigned int *num_blocks_, unsigned int *num_std_wells_, unsigned int *dim_, unsigned int *dim_wells_){
+    *num_blocks_ = num_blocks;
+    *num_std_wells_ = num_std_wells;
+    *dim_ = dim;
+    *dim_wells_ = dim_wells;
+}
 
-    // apply StandardWells
-    if (num_std_wells > 0) {
-        OPM_THROW(std::logic_error, "Error StandardWells are not supported by openclSolver");
-    }
+void WellContributions::getData(double **valsC, double **valsD, double **valsB, int **colsC, int **colsB, unsigned int **val_pointers_){
+    std::copy(d_Cnnzs_ocl, d_Cnnzs_ocl + num_blocks*dim*dim_wells, *valsC);
+    std::copy(d_Dnnzs_ocl, d_Dnnzs_ocl + num_std_wells*dim_wells*dim_wells, *valsD);
+    std::copy(d_Bnnzs_ocl, d_Bnnzs_ocl + num_blocks*dim*dim_wells, *valsB);
+    std::copy(d_Ccols_ocl, d_Ccols_ocl + num_blocks, *colsC);
+    std::copy(d_Bcols_ocl, d_Bcols_ocl + num_blocks, *colsB);
+    std::copy(val_pointers, val_pointers + num_std_wells + 1, *val_pointers_);
 }
+
 #endif
 
-void WellContributions::addMatrix([[maybe_unused]] MatrixType type, [[maybe_unused]]int *colIndices, [[maybe_unused]] double *values, [[maybe_unused]] unsigned int val_size)
+void WellContributions::addMatrix(MatrixType type, int *colIndices, double *values, unsigned int val_size)
```

Comment:
Why did you remove the `[[maybe_unused]]` specifiers? They still are unused if we do not support the GPU, aren't they?

Reply:
Again, no specific reason. I think I removed them to run some tests and forgot to put them back. Already fixing it.